### PR TITLE
Feat:Added like and dislike feature and refactor code

### PR DESCRIPTION
**Pull request by tushargoyal812**

### CODE REVIEW

File: src/components/navbar/navbar.js [hunk 1]
```diff
@@ -9,21 +9,21 @@ export const Navbar=()=>{
...
-        <header class="e-com-header">
-        <nav class="e-com-navbar shop-nav ecom-bg-blue">
+        <header className="e-com-header">
+        <nav className="e-com-navbar shop-nav ecom-bg-blue">
             <Link to="/">
-                <div class="ecom-white">showTube</div>
+                <div className="ecom-white">showTube</div>
             </Link>
-            <ul class="e-com-nav-items">
+            <ul className="e-com-nav-items">
```
Comment (sharmayatendra) [inline src/components/navbar/navbar.js:17]: you can change class name as it is not "e-com" looks odd for reviewer

File: src/components/navbar/navbar.js [hunk 2]
```diff
@@ -9,21 +9,21 @@ export const Navbar=()=>{
...
                 <Link to="/videolisting">videos</Link>
             </ul>
-            <input type="text" class="search-input ecom-search" placeholder="search" />
-            <div class="e-com-social">
-                <div class="ecom-login-container">
+            <input type="text" className="search-input ecom-search" placeholder="search" />
+            <div className="e-com-social">
+                <div className="ecom-login-container">
```
Comment (sharmayatendra) [inline src/components/navbar/navbar.js:22]: same here can use generic names

File: src/components/sidebar/sidebar.js [hunk 3]
```diff
@@ -2,27 +2,27 @@ import './sidebar.css'
...
     return(
-        <aside class="product-sidebar p-2">
-            {/* <div class="filter p-1">
+        <aside className="product-sidebar p-2">
+            {/* <div className="filter p-1">
                 <h2>filter</h2> <a href="">clear</a>
             </div> */}
-            <Link to="/" class="price p-1">
-                <h2 class="">Home</h2>
+            <Link to="/" className="price p-1">
```
Comment (sharmayatendra) [inline src/components/sidebar/sidebar.js:9]: same as above

File: src/context/playlist-context.js [hunk 4]
```diff
@@ -6,7 +6,11 @@ const PlaylistContext=createContext()
 
 const PlaylistProvider=({children})=>{
     const [playlist,setPlaylist]=useState([])
-    return (<PlaylistContext.Provider value={{playlist,setPlaylist}}>{children}</PlaylistContext.Provider>)
+    const [playlistName,setPlaylistName]=useState(false)
+    const [playlistInput,setPlaylistInput]=useState()
+    const [modal,setModal]=useState(false)
+    const [sentPlayList,setSentPlayList]=useState(null)
```
Comment (sharmayatendra) [inline src/context/playlist-context.js:12]: since states are more than 3 you can go for reducer

File: src/index.js [hunk 5]
```diff
@@ -7,12 +7,14 @@ import {BrowserRouter} from 'react-router-dom'
...
 import { VideoProvider } from "./context/video-context";
 import { PlaylistProvider } from "./context/playlist-context";
+import { LikeProvider } from "./context/like-context";
 
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
+    <LikeProvider>
```
Comment (sharmayatendra) [inline src/index.js:17]: i think  browser router should be on top of other providers

**Review by sharmayatendra — APPROVED** [review]
good file structure
used component for everything thats good
you should add screenshots if not deployed preview

suggested few changes you can look into that

File: src/components/navbar/navbar.js (hunk 1 above)
Comment (tushargoyal812) [reply src/components/navbar/navbar.js:17]: ok will do it

File: src/components/navbar/navbar.js (hunk 2 above)
Comment (tushargoyal812) [reply src/components/navbar/navbar.js:22]: ok

File: src/components/sidebar/sidebar.js (hunk 3 above)
Comment (tushargoyal812) [reply src/components/sidebar/sidebar.js:9]: ok

File: src/context/playlist-context.js (hunk 4 above)
Comment (tushargoyal812) [reply src/context/playlist-context.js:12]: will change it in final version.

File: src/index.js (hunk 5 above)
Comment (tushargoyal812) [reply src/index.js:17]: ok will change it

File: src/components/modal/modal.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+                            <span>{item.title}</span>
+                        </div>
+                    ))}
+                </div>
+                {playlistName?<div>
+                    <label htmlFor="">Name</label>
+                    <input onChange={(e)=>setPlaylistInput(e.target.value)} type="text" placeholder='enter playlist name' />
+                </div>:""}
+                <hr />
+                <div className="modal-footer">
```
Comment (vishalg8454) [inline src/components/modal/modal.js]: && can be used instead of ?

File: src/pages/liked-videos/liked-videos.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        <div className="playlist-container">
+            {like.map(item=>(
+                <div className="playlist-details">
+                    <img className="playlist-image" src={item.image} />
+                    <div>
+                    <h2 className="playlist-name mx-1">{item.description} </h2>
+                    <div onClick={()=>dislikedVideoHandler(item,setLike)} className='dislike-in-like flex mt-2'>
+                    <span class="material-icons-outlined">thumb_down</span>
+                    <span>DISLIKE</span>
+                    </div>
```
Comment (vishalg8454) [inline src/pages/liked-videos/liked-videos.js]: `key` prop is missing

File: src/util-functions/watch-later.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import axios from "axios";
+export const watchLaterHandler= async (video,setWatchLater)=>{
+    const token=localStorage.getItem("user")
+    console.log(video);
+    console.log(token);
+    try{
+        const response=await axios.post("/api/user/watchlater",
+    {
```
Comment (vishalg8454) [inline src/util-functions/watch-later.js]: remove `console.logs()`

**Review by vishalg8454 — APPROVED** [review]
All looks goods, just minor change.

File: src/components/modal/modal.js (hunk 6 above)
Comment (tushargoyal812) [reply src/components/modal/modal.js]: yes will do it

File: src/pages/liked-videos/liked-videos.js (hunk 7 above)
Comment (tushargoyal812) [reply src/pages/liked-videos/liked-videos.js]: yes will add it

File: src/util-functions/watch-later.js (hunk 8 above)
Comment (tushargoyal812) [reply src/util-functions/watch-later.js]: ok

**Pull request merged**